libs/SalesforceSDKCore/SalesforceSDKCore/Classes/SFAppStyler.m

```diff
@@ -0,0 +1,210 @@
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "SFAppStyler.h"
+
+@interface SFAppStyler ()
+
+@property (nonatomic, strong) NSMutableDictionary *fontMap;
+
+@end
+
+@implementation SFAppStyler
+
++(instancetype)sharedInstance {
+    static dispatch_once_t onceToken;
+    static SFAppStyler *styler = nil;
+    dispatch_once(&onceToken, ^{
+        styler = [[self alloc] init];
+    });
+    return styler;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        _fontMap = [[NSMutableDictionary alloc] init];
+        
+        // default values
+        _titleColor = [UIColor colorWithRed:0.024 green:0.110 blue:0.245 alpha:1.0];
+        _primaryAppColor = [UIColor colorWithRed:22.0/255.0 green:87.0/255.0 blue:205/255.0 alpha:1.0];
+        _actionableTextColor = [UIColor colorWithRed:0.333 green:0.525 blue:0.937 alpha:1.0];
+        _headerTextColor = [UIColor whiteColor];
+        _primaryTextColor = [UIColor blackColor];
+        _toolbarTextColor = nil;
+        _toolbarTextFont = nil;
+        [self updateDefaultSize];
+    }
+    return self;
+}
+
+- (void)updateDefaultSize {
+    self.headerFont =  [self fontWithSize:16.0 style:SFAppStylerFontStyleBold];
+    self.primaryTextFont = [self fontWithSize:14.0 style:SFAppStylerFontStyleRegular];
+    self.actionableFont = [self fontWithSize:14.0 style:SFAppStylerFontStyleRegular];
+}
+
+- (void)setFontName:(NSString * _Nullable)fontName forStyle:(SFAppStylerFontStyle)style {
+    if (fontName) {
+        self.fontMap[@(style)] = fontName;
+    } else {
+        [self.fontMap removeObjectForKey:@(style)];
+    }
+    [self updateDefaultSize];
+}
+
+- ( UIFont * _Nonnull )fontWithSize:(CGFloat)fontSize style:(SFAppStylerFontStyle)style {
+    NSString *fontToUse = self.fontMap[@(style)];
+    if (fontToUse) {
+        return [UIFont fontWithName:fontToUse size:fontSize];
+    } else {
+        switch (style) {
+            case SFAppStylerFontStyleBold:
+                return [UIFont boldSystemFontOfSize:fontSize];
+                break;
+            default:
+                return [UIFont systemFontOfSize:fontSize];
+                break;
+        }
+    }
+}
+
+#pragma mark - Styling Methods
+
+- (nonnull UIImage *)headerBackgroundImage {
```

Comment:
Is there a particular benefit to creating the background color of a navigation bar as an image, rather than simply setting `UINavigationBar.barTintColor`?  Seems like the latter would be more efficient.